src/mono/driver_blazor.c

```diff
@@ -1,65 +1,77 @@
 static void*
-blazor_invoke_js (MonoString **exceptionMessage, MonoString *funcName, void* arg0, void* arg1, void* arg2)
+mono_wasm_invoke_js_marshalled (MonoString **exceptionMessage, void *asyncHandleLongPtr, MonoString *funcName, MonoString *argsJson)
 {
 	*exceptionMessage = NULL;
 	char *funcNameUtf8 = mono_string_to_utf8 (funcName);
-	void *jsCallResult = (void *)EM_ASM_INT ({
+	char *argsJsonUtf8 = argsJson == NULL ? NULL : mono_string_to_utf8 (argsJson);
+
+	MonoString *resultJsonMonoString = (void *)EM_ASM_INT ({
```

Comment:
So I am assuming that this block is 'inline assembly' in the form of JS.

S P I C Y

Reply:
Exactly: https://kripken.github.io/emscripten-site/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#interacting-with-code-call-javascript-from-native